src/TestingUtils/Microsoft.AspNetCore.Analyzer.Testing/src/DiagnosticProject.cs

```diff
@@ -23,24 +24,35 @@ public class DiagnosticProject
         /// </summary>
         public static string TestProjectName = "TestProject";
 
+        private static readonly Dictionary<Assembly, Solution> _solutionCache = new Dictionary<Assembly, Solution>();
```

Comment:
Does this need to be `Concurrent`? xunit runs tests across classes in parallel?

Reply:
That's why I lock around access.